packages/terra-demographics-banner/src/DemographicsBanner.jsx

```diff
@@ -0,0 +1,182 @@
+  /**
+   * The persons preferred first name if they have one.
+   */
+  preferredFirstName: PropTypes.string,
+};
+
+const defaultProps = {
+  additionalDetails: null,
+  age: null,
+  applicationRows: [],
+  dateOfBirth: null,
+  deceasedDate: null,
+  gender: null,
+  gestationalAge: null,
+  identifiers: {},
+  personName: null,
+  photo: null,
+  postMenstrualAge: null,
+  preferredFirstName: null,
+};
+
+class DemographicsBanner extends React.Component {
+  additionalApplicationRows() {
+    return this.props.applicationRows.map((row, index) => (
+      // eslint-disable-next-line react/no-array-index-key
+      <div key={`additional-row-${index}`} className="terra-DemographicsBanner--row">
+        {Object.keys(row).map(key => (
+          <span key={`additional-row-${key}-${row[key]}`}>{key}: <b>{row[key]}</b></span>
+        ))}
+      </div>
+    ));
+  }
+
+  personDetails() {
+    const noDataMessage = '--';
+    const dateOfBirth = this.props.dateOfBirth || `DOB: ${noDataMessage}`;
+
+    const elements = [
+      <span key="age"><b>{ this.props.age || noDataMessage }</b></span>,
+      <span key="gender"><b>{ this.props.gender || noDataMessage }</b></span>,
+      <span key="dob" dangerouslySetInnerHTML={{ __html: dateOfBirth }} />,
+    ];
+
+    if (this.props.gestationalAge) {
+      elements.push(<span key="ga" dangerouslySetInnerHTML={{ __html: this.props.gestationalAge }} />);
+    }
+
+    if (this.props.postMenstrualAge) {
+      elements.push(<span key="pma" dangerouslySetInnerHTML={{ __html: this.props.postMenstrualAge }} />);
+    }
+
+    if (this.props.deceasedDate) {
+      elements.push(<span key="deceased" dangerouslySetInnerHTML={{ __html: this.props.deceasedDate }} />);
+    }
+
+    return elements;
+  }
+
+  applicationIdentifiers() {
+    const identifiers = this.props.identifiers;
+
+    if (identifiers) {
+      return Object.keys(identifiers).map(key =>
+        <span key={`identifier-${key}`}>{key}: <b>{identifiers[key]}</b></span>,
+      );
+    }
+
+    return null;
+  }
+
+  render() {
+    const {
+      age,
+      additionalDetails,
+      applicationRows,
+      dateOfBirth,
+      deceasedDate,
+      gender,
+      gestationalAge,
+      identifiers,
+      personName,
+      photo,
+      postMenstrualAge,
+      preferredFirstName,
+      ...customProps
+    } = this.props;
+
+    const mainClasses = classNames(
+      'terra-DemographicsBanner',
+      { 'terra-DemographicsBanner--deceased': deceasedDate },
+      customProps.className,
+    );
+
+    delete customProps.className;
+
+    return (
+      <section className={mainClasses} {...customProps}>
+        <div className="terra-DemographicsBanner--profile-photo">
+          {this.props.photo}
+        </div>
+        <div className="terra-DemographicsBanner--content">
+          <div className="terra-DemographicsBanner--row">
+            <h1 className="terra-DemographicsBanner--person-name">
+              { personName || '--' }
```

Comment:
I am curious-here set the default props as null and give that particular prop a value if one wasn't passed in. Is this better to assign a value when one isn't present instead of assigning a default?